tests/conftest.py

```diff
@@ -202,6 +202,67 @@ def _generate_work_chain(entry_point, inputs=None):
     return _generate_work_chain
 
 
+@pytest.fixture
+def generate_calcjob_node():
```

Comment:
This fixture looks pretty familiar :D
Maybe we could merge it with `generate_calculation_node`, providing an additional input argument for when a `CalcJob` is
being created to avoid duplication? Not sure if there are any other `CalcJob` specific changes in this implementation?

Reply:
yeah.. we could do that as well, I just thought maybe this way makes sense as in future we could add more features for this fixture... but yeah idk

Reply:
As agreed during in-person discussion, we'll remove the currently duplicated code (from the `generate_calculation_node` fixture which I needed for testing the dumping functionality), as it's not really needed here, and instead keep the `generate_calcjob_node` fixture at the minimum needed to test the changes of this PR.